### ISSUE TITLE
Key handling issue

### ISSUE THREAD

**Issue by smblott-github** [philc/vimium#2453]
From the Chrome store...

![snapshot](https://cloud.githubusercontent.com/assets/2641335/24078826/4e482f5c-0c70-11e7-9f20-c40171c891d1.png)

**Comment by gdh1995** [comment]
@smblott-github  I believe this strange bug is caused by custom events:
* the host scripts creates any other type of events, like `new UIEvent('keydown')`
* then this event is dispatched to window, and Vimium breaks.

I've tested the case above, and the error info just matches exactly.

So we doesn't need to fix it. And if you want, it's a better choice to check `event.isTrusted`

**Comment by smblott-github** [comment]
Thanks for that, @gdh1995.  Yes, checking `event.isTrusted` seems like a better idea (if your theory is correct).

**Comment by derrotebaron** [comment]
You REALLY should check isTrusted. Otherwise the Webpage is able to issue any command bound to any key.

Short Example:
```html
<!DOCTYPE html>

<html>
    <head>
        <script>
         setTimeout(function (){
             document.dispatchEvent(new KeyboardEvent('keydown', { key: 'y' }))
             document.dispatchEvent(new KeyboardEvent('keydown', { key: 'y' }))
             while (true) {
                 document.dispatchEvent(new KeyboardEvent('keydown', { key: 'P' }));
             }
         }, 1000);
        </script>
    </head>
	  <body>
	  </body>
</html>
```
This is more or less a forkbomb opening many tabs. There are possibly more ways to exploit this bug, potentially causing information leaks.

Please check for isTrusted on any event.

(This is the same bug as https://github.com/1995eaton/chromium-vim/issues/572)

**Comment by mrmr1993** [comment]
This can be closed now #2605 has been merged.
